integration/watch.ts

```diff
@@ -1,86 +1,150 @@
+    $ yarn watch --polling
+    $ yarn watch simple struct
+        `);
+  process.exit(0);
+}
 
-    chokidar
-      .watch("*/*.proto", { ignoreInitial: true, usePolling  })
-      .on('all', yarnRunHandler(yarn, 'proto2bin'));
+function showSettings(tests: string[]) {
+  console.log(`${colors.yellow}`);
+  console.log(`Watching for changes:`);
+  console.log('- integration/\tRecompiling .proto to .bin and .ts');
+  if (tests.length) {
+    console.log(`- src/\t\tRecompiling .proto to .ts\n\t\t(${tests.join(', ')})`);
+  }
+  console.log('\nPress enter to recompile all .proto files.');
+  console.log('Press ctrl+c or q to exit');
+  console.log(colors.reset);
+}
 
-    chokidar
-      .watch("*/*.proto", { ignoreInitial: true, usePolling })
-      .on('all', yarnRunHandler(yarn, 'proto2pbjs'));
+function integrationHandler(yarn: string, task: string) {
+  return (event: FsEvent, triggerPath: string) => {
+    if (event !== 'add' && event !== 'change') {
+      return;
+    }
 
-    chokidar
-      .watch("*/*.bin", { ignoreInitial: true, usePolling })
-      .on('all', yarnRunHandler(yarn, 'bin2ts'));
-}
+    triggerPath = triggerPath.replace(/\\/g, '/'); // windows
+    const relativePath = triggerPath.replace(/^integration\//, '');
 
-const colors = {
-    none: '',
-    reset: '\x1b[0m',
-    green: '\x1b[32m',
-    red: '\x1b[31m',
-    yellow: '\x1b[33m',
-    blue: '\x1b[34m',
-    magenta: '\x1b[35m',
-    cyan: '\x1b[36m'
+    yarnRun(yarn, task, triggerPath, relativePath);
+  };
 }
 
-function yarnRunHandler(yarn: string, task: string) {
-    return (event: "add" | "addDir" | "change" | "unlink" | "unlinkDir", path: string) => {
-        if (event !== 'add' && event !== 'change') {
-            return;
-        }
+function yarnRun(yarn: string, task: string, header: string, taskArgument?: string) {
+  const yarnArgs = taskArgument ? [task, taskArgument] : [task];
 
-        path = path.replace(/\\/g, "/"); // windows
+  console.log(formatLog(colors.green, header, task, `${yarn} ${yarnArgs.join(' ')}`));
 
-        yarnRun(yarn, task, path);
+  const yarnProcess = spawn(yarn, yarnArgs);
+  yarnProcess.stdout.on('data', (data: Buffer) => console.log(formatLog(colors.none, header, task, data.toString())));
+  yarnProcess.stderr.on('data', (data: Buffer) => console.error(formatLog(colors.red, header, task, data.toString())));
+  yarnProcess.on('error', (err: Error) => console.error(formatLog(colors.red, header, task, err.message)));
+  yarnProcess.on('close', (code: number) => {
+    if (code !== 0) {
+      console.error(formatLog(colors.red, header, task, `Exited with code ${code}`));
+    } else {
+      console.log(formatLog(colors.green, header, task, 'Done'));
     }
+  });
 }
 
-function yarnRun(yarn: string, task: string, path: string) {
-    const yarnArgs = [task, path];
-
-    console.log(formatLog(colors.green, task, path, `${yarn} ${yarnArgs.join(' ')}`));
+function sourceHandler(yarn: string, task: string, tests: string[]) {
```

Comment:
Maybe `srcHandler` to highlight its handling the `src/` dir?

Reply:
I tend to think full names are more clear than abbreviations, but sure :-)